tests/checker/__init__.py

```diff
@@ -119,6 +119,8 @@ def end_output (self, linknumber=-1, **kwargs):
         """
         Stores differences between expected and result in self.diff.
         """
+        self.expected.sort()
+        self.result.sort()
```

Comment:
Um.  This would be fine if self.expected and self.result had one line per test URL, but from what I can see these reports have little details like "line number 42" all in separate items.

This may make it hard to debug differences when you're looking at a unified diff of test failures.

Reply:
This approach could work, I think, if you did something like

```
self.expected = self.normalize(self.expected)
self.result = self.normalize(self.result)
```

with

```python
def normalize(self, result_log):
    # XXX we assume that each log entry has a URL key, maybe we should add an assert into log_url() to that effect?
    sep = '\nurl '
    return sep.join(sorted('\n'.join(result_log).split(sep)))
```

Reply:
That works for me. Updated the branch.